community/values/src/main/java/org/neo4j/values/Values.java

```diff
@@ -0,0 +1,354 @@
+    public static Value lazyBooleanArray( ValueLoader<boolean[]> producer )
+    {
+        return new BooleanArray.Lazy( producer );
+    }
+
+    public static Value numberValue( Number number )
+    {
+        if ( number instanceof Long )
+        {
+            return longValue( number.longValue() );
+        }
+        if ( number instanceof Integer )
+        {
+            return intValue( number.intValue() );
+        }
+        if ( number instanceof Double )
+        {
+            return doubleValue( number.doubleValue() );
+        }
+        if ( number instanceof Byte )
+        {
+            return byteValue( number.byteValue() );
+        }
+        if ( number instanceof Float )
+        {
+            return floatValue( number.floatValue() );
+        }
+        if ( number instanceof Short )
+        {
+            return shortValue( number.shortValue() );
+        }
+
+        throw new UnsupportedOperationException( "Unsupported type of Number " + number.toString() );
+    }
+
+    public static Value longValue( long value )
+    {
+        return new LongValue( value );
+    }
+
+    public static Value intValue( int value )
+    {
+        return new IntValue( value );
+    }
+
+    public static Value shortValue( short value )
+    {
+        return new ShortValue( value );
+    }
+
+    public static Value byteValue( byte value )
+    {
+        return new ByteValue( value );
+    }
+
+    public static Value booleanValue( boolean value )
+    {
+        return new BooleanValue( value );
+    }
+
+    public static Value charValue( char value )
+    {
+        return new CharValue( value );
+    }
+
+    public static Value doubleValue( double value )
+    {
+        return new DoubleValue( value );
+    }
+
+    public static Value floatValue( float value )
+    {
+        return new FloatValue( value );
+    }
+
+    public static Value stringArray( String[] value )
+    {
+        return new StringArray.Direct( value );
+    }
+
+    public static Value byteArray( byte[] value )
+    {
+        return new ByteArray.Direct( value );
+    }
+
+    public static Value longArray( long[] value )
+    {
+        return new LongArray.Direct( value );
+    }
+
+    public static Value intArray( int[] value )
+    {
+        return new IntArray.Direct( value );
+    }
+
+    public static Value doubleArray( double[] value )
+    {
+        return new DoubleArray.Direct( value );
+    }
+
+    public static Value floatArray( float[] value )
+    {
+        return new FloatArray.Direct( value );
+    }
+
+    public static Value booleanArray( boolean[] value )
+    {
+        return new BooleanArray.Direct( value );
+    }
+
+    public static Value charArray( char[] value )
+    {
+        return new CharArray.Direct( value );
+    }
+
+    public static Value shortArray( short[] value )
+    {
+        return new ShortArray.Direct( value );
+    }
+
+    // BOXED FACTORY METHODS
+
+    public static Value of( Object value )
```

Comment:
how about making the if-tree in here smaller by checking for `value instanceof Number` (instead of all the individual number types) and delegate to the `numberValue` method (which should be moved to this _"BOXED FACTORY METHODS"_ section by the way)?